apps/src/music/views/MusicView.jsx

```diff
@@ -276,6 +289,30 @@ class UnconnectedMusicView extends React.Component {
     this.musicBlocklyWorkspace.updateHighlightedBlocks(playingBlockIds);
   };
 
+  loadProgressionStep = async () => {
+    let progressionStep = undefined;
+    if (this.props.levels) {
+      // Since we have levels, we'll asynchronously retrieve the current level data.
+      const response = await this.loadLevelData();
+      progressionStep = response.level_data;
+
+      // Also, just set the number of levels.
+      this.setState({levelCount: this.props.levels.length});
```

Comment:
I moved it to the `musicRedux` store and did a bunch of refactoring in nearby code.  It is a bit tricky that we are now supporting four distinct cases: standalone with no progression, standalone with a progression file, as a script level, and as a level.  I didn't yet find a clean way to set `levelCount` only once for each case (because loading & reloading use common code), but can take another look soon.